tests/TestCase/Console/ConsoleOptionParserTest.php

```diff
@@ -330,6 +332,38 @@ public function testAddMultipleOptionsWithMultiple()
         $this->assertEquals($expected, $result[0], 'options with multiple values did not parse');
     }
 
+    /**
+     * test adding a required option.
+     *
+     * @return void
+     */
+    public function testAddOptionRequired()
+    {
+        $parser = new ConsoleOptionParser('test', false);
+        $parser
+            ->addOption('test', [
+                'default' => 'default value',
+                'required' => true,
+            ])
+            ->addOption('no-default', [
+                'required' => true,
+            ]);
+        $result = $parser->parse(['--test', '--no-default', 'value']);
+        $this->assertSame(
+            ['test' => 'default value', 'no-default' => 'value', 'help' => false],
+            $result[0]
+        );
+
+        $result = $parser->parse(['--no-default', 'value']);
+        $this->assertSame(
+            ['no-default' => 'value', 'help' => false, 'test' => 'default value'],
+            $result[0]
+        );
+
+        $this->expectException(ConsoleException::class);
```

Comment:
If the first parse throws an exception, does phpunit mark the remaining asserts as fails?

Reply:
Anything that happens before expectException should still get counted as passed. Nothing will happen after the exception is raised though.

Reply:
I prefer to have exception tests in a separate function to avoid capturing it for other tests so nothing is skipped.

Reply:
Ok. I can split them up.